### PR TITLE
HBHE: SiPM pulse shape 207 for 2018data

### PR DESCRIPTION
**Pull request by hmiaozh**
A new SiPM pulse shape (shape 207) for 2018data.
Derived from data of 2017 PhaseScan.
Details in the slides: https://indico.cern.ch/event/677823/contributions/2775510/attachments/1560954/2457326/Shape207_test.pdf

Note: I'll update after PR #21168 is merged.

### CODE REVIEW

**Comment by perrotta** [comment]
please test

File: CalibCalorimetry/HcalAlgos/src/HcalPulseShapes.cc [hunk 1]
```diff
@@ -268,6 +270,283 @@ void HcalPulseShapes::computeHFShape() {
...
+   //Combination of all phase scan data (May,Jul,Oct2017)
+   //L(X)AG
+   //runs:  294736-294740, 294929-294950, 298594-298598 and 305744-305758
+
+  unsigned int nbin = 250;
+
+  std::vector<float> nt = {
+	5.22174e-12,
+	7.04852e-10,
+	3.49584e-08,
```
Comment (slava77) [inline CalibCalorimetry/HcalAlgos/src/HcalPulseShapes.cc]: please wrap multiple entries on one line (e.g. 10)
also,  can this  be changed to an std::array?

Comment (hmiaozh) [reply CalibCalorimetry/HcalAlgos/src/HcalPulseShapes.cc]: All the three related templates are changed to type std::array with multiple entries per line. Please check. Let me know if you have any comment.

**Comment by perrotta** [comment]
please test

**Comment by civanch** [comment]
In RecoLocalCalo/HcalRecAlgos/src/HcalDeterministicFit.cc  computations are in float, so constants should be float as well to avoid frequent convertion between types: 0.1 -> 0.1f, 0 -> 0.f.

**Comment by mariadalfonso** [comment]
@civanch 
I fixed the constant floats in my cleanup PR https://github.com/cms-sw/cmssw/pull/21289/commits/7d57416908aee97145ad293f852ad9bc6a0f859a

**Comment by civanch** [comment]
+1

**Comment by slava77** [comment]
@hmiaozh 
This is just to clarify on the status of review from reco:
changes look OK except in places where a different implementation is in place in #21168. 
So, the code will need to be updated. You may also choose to merge two PRs.

**Comment by mariadalfonso** [comment]
@slava77 
Would be easier if @cmadrid1 cherry pick the commit from @hmiaozh and merge into the #21168

**Comment by slava77** [comment]
-1 

superseded by #21168

**Pull request closed without merging**